### DISCUSSION TITLE
Migration from ZenHub - Initial Feedback for our Kanban workflow

### DISCUSSION THREAD

**Discussion by lfrancke** [community/community#31227] [Projects and Issues]
Hi,

we've been part of the beta since sometime in the middle and have tried to use the new Projects multiple times and always had to abort. We're a fully Open Source shop and are using ZenHub so far. We are using a Kanban workflow and, unfortunately, we found that the new projects experience is not a great match for such a workflow so far.

@glortho[ asked me](https://github.com/orgs/community/discussions/30707#discussioncomment-3483222) to start a feedback thread over here so I'll list my main pain points here full well knowing that there are already dedicated threads to some of these

- ~~Make it possible to link a PR to an issue in a different repository~~
  - ~~Currently only links to issues in the same repository are possible~~
- Some way to tackle hierarchical tasks (Epics -> Stories -> Tasks)
  - Ideally using workflow integration (e.g. "when an Epic has all Stories finished move it to X")
- Show more details about "nested" PRs (e.g. reviewers) so we only have to add one of the two to the board (PR and/or issue) or at least be able to move them around quickly together
  - In ZenHub a linked PR is "nested" within an issue and both can be moved at the same time
- Swimlanes within a board view (e.g. an "Expedite" lane)
- Mark an issue as blocked by another one including visualization (like the current "Reviewers" badge for PRs)
- Better/easier (i.e. no Github Actions)/more powerful workflows:
  - A way to add new issues from repositories automatically to a project (for triaging)
  - A way to automatically add/move issues to another project when it enters a column (e.g. an item in a "Done" column in a "Design" project should be added to the "Development" project automatically)
- Project events on issues/PRs to identify when an issue has moved into a state
- A description/label/sticker per column that can be shown at the top (we use this for WIP limit display, entry & exit rules, definition of done etc.)
- WIP limits (at least visualize them)
- Split columns (e.g. a "Development" column split into "Waiting For", "In Progress" and "Done")
- A way to move items between columns in board view using a small dropdown instead of dragging & dropping
- "Send to top" and "Send to bottom" in a dropdown menu on cards
- Allow adding items from other organizations/repositories
- ~~Option to hide empty columns in board views~~
- Sharing of fields across projects
- Allow "OR" filters or extend the 256 char limit

Some of these are more important than others for our workflow.
I tried to roughly sort them by priority (top = most important).
Happy to go into more details on any of these and if it helps I can make our board public.

Ideally, we'd be able to use a single board targeted at "Product Owners" with a high level view and at Developers going down to the low level. We're currently using multiple projects for this (in ZenHub) but it's hard to keep that in sync. This is not yet possible/easy to do in Github projects but I hope we'll get there.

I've written a rudimentary command line tool that uses the GraphQL API to get the stats I need so I can alert on WIP limits etc. but the more we can get in the UI the better.

**Comment by azenMatt** [comment]
Thanks so much for sharing this @lfrancke 

A couple of comments... 

> Make it possible to link a PR to an issue in a different repository

This is being worked on at the moment 😄 

> Some way to tackle hierarchical tasks (Epics -> Stories -> Tasks)

So is this! 

> Better/easier (i.e. no Github Actions)/more powerful workflows:
> A way to add new issues from repositories automatically to a project (for triaging)

And this one

> Project events on issues/PRs to identify when an issue has moved into a state

Also this!

> Option to hide empty columns in board views

This will be live later today.... 


So sounds like we are in sync with where we both what projects to go 😄 All your other feedback is very valid as well and work we will get to!

**Reply by lfrancke** [reply]
That sounds awesome! Thank you for the update.
You have me excited for later today then :)

If there is anything we can do to help please let me know.

**Reply by lfrancke** [reply]
It looks like issues and PRs can now be linked from other repositories as well. That's excellent!

**Reply by lfrancke** [reply]
@azenMatt I see that the hierarchy feature is in private beta or will start soon (https://github.com/github/roadmap/issues/284): If you need more volunteers we'd love to join.

**Reply by azenMatt** [reply]
I just shared this with the Hierarchy PM to reach out 😄

**Reply by alexanderniebuhr** [reply]
@azenMatt please let the PM also know, we would like to get in contact for our company to discuss private beta access.
